libraries/LittleFS/src/LittleFS.h

```diff
@@ -0,0 +1,446 @@
+        _lfs_cfg.lookahead = 256;
+        _lfs_cfg.read_buffer = nullptr;
+        _lfs_cfg.prog_buffer = nullptr;
+        _lfs_cfg.lookahead_buffer = nullptr;
+        _lfs_cfg.file_buffer = nullptr;
+    }
+
+    ~LittleFSImpl() {
+        if (_mounted) {
+            lfs_unmount(&_lfs);
+        }
+    }
+
+    FileImplPtr open(const char* path, OpenMode openMode, AccessMode accessMode) override;
+    DirImplPtr openDir(const char *path) override;
+
+    bool exists(const char* path) override {
+        lfs_info info;
+	if (!_mounted || !path[0]) {
+            return false;
+        }
+        char lfsPath[LFS_NAME_MAX+1];
+        int rc = lfs_stat(&_lfs, spiffsNameToLFS(path, lfsPath), &info);
+        return rc == 0;
+    }
+
+    bool rename(const char* pathFrom, const char* pathTo) override {
+        if (!_mounted) {
+            return false;
+        }
+        int rc = lfs_rename(&_lfs, pathFrom, pathTo);
+        if (rc != 0) {
+            DEBUGV("lfs_rename: rc=%d, from=`%s`, to=`%s`\n", rc, pathFrom, pathTo);
+            return false;
+        }
+        return true;
+    }
+
+    bool info(FSInfo& info) override {
+        if (!_mounted) {
+            return false;
+        }
+        info.maxOpenFiles = _maxOpenFds;
+        info.blockSize = _blockSize;
+        info.pageSize = _pageSize;
+        info.maxOpenFiles = _maxOpenFds;
+        info.maxPathLength = LFS_NAME_MAX;
+        info.totalBytes = _size;
+        info.usedBytes = _size - _getUsedBlocks() * _blockSize;
+        return true;
+    }
+
+    bool remove(const char* path) override {
+        if (!_mounted) {
+            return false;
+        }
+        int rc = lfs_remove(&_lfs, path);
+        if (rc != 0) {
+            DEBUGV("lfs_remove: rc=%d path=`%s`\n", rc, path);
+            return false;
+        }
+        return true;
+    }
+
+    bool begin() override {
+        if (_size <= 0) {
+            DEBUGV("LittleFS size is <= zero");
+            return false;
+        }
+        if (_tryMount()) {
+            return true;
+        }
+        if (!format()) {
+            return false;
+        }
+        return _tryMount();
+    }
+
+    void end() override {
+        if (!_mounted) {
+            return;
+        }
+        lfs_unmount(&_lfs);
+        _mounted = false;
+    }
+
+    bool format() override {
+        if (_size == 0) {
+            DEBUGV("lfs size is zero\n");
+            return false;
+        }
+
+        bool wasMounted = _mounted;
+        if (_mounted) {
+            lfs_unmount(&_lfs);
+            _mounted = false;
+        }
+
+        memset(&_lfs, 0, sizeof(_lfs));
+        int rc = lfs_format(&_lfs, &_lfs_cfg);
+        if (rc != 0) {
+            DEBUGV("lfs_format: rc=%d\n", rc);
+            return false;
+        }
+
+        if (wasMounted) {
+            return _tryMount();
+        }
+
+        return true;
+    }
+
+    // Because "/" is an actual full-fledged dir separator in LFS, we need to
```

Comment:
I understand the idea, but not the specifics. What is the purpose of the reserved char? Does littlefs interpret it in some special way?

Reply:
Removed, you shamed me into using real directories and not faking SPIFFS flat filesystem.  There'll be new code replacing this in the next push.